src/main/java/pl/orange/NextDoorBook/exchange/ExchangeService.java

```diff
@@ -22,6 +31,31 @@ public ExchangeAddDTO addExchange(ExchangeAddDTO exchangeAddDTO) {
 
     }
 
+    public ExchangeDTO addBookReservation(ExchangeReservationDTO exchangeReservationDTO) {
+        Book bookToExchange = bookRepository.getBookByID(exchangeReservationDTO.bookId())
+                .orElseThrow(() ->
+                        new BookNotFoundException("Book with id " + exchangeReservationDTO.bookId() + "does not exist"));
+        User renter = userRepository.getUserById(exchangeReservationDTO.renterId())
+                .orElseThrow(() ->
+                        new UserNotFoundException("User with id " + exchangeReservationDTO.renterId() + " does not exist"));
+
+        Exchange reservationToSave = createReservation(bookToExchange, renter);
+        return exchangeDTOMapper.mapToDTO(exchangeRepository.saveExchange(reservationToSave));
+    }
+
+    private Exchange createReservation(Book bookToExchange, User renter) {
```

Comment:
DO przemyślenia, czy nie można tego zrobić zamiast encji na DTO

Reply:
chcesz zrobić inaczej to zrób. Ja tak zostawiam :D